test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ViewComponents/DefaultViewComponentSelectorTest.cs

```diff
@@ -112,6 +112,26 @@ public void SelectComponent_Ambiguity()
             Assert.Equal(expected, ex.Message);
         }
 
+        [Theory]
+        [InlineData("Name")]
+        [InlineData("Ambiguous.Name")]
+        public void SelectComponent_AmbiguityDueToDerivation(string name)
+        {
+            var selector = CreateSelector();
+            var expected =
+                $"The view component name '{name}' matched multiple types:" + Environment.NewLine +
+                $"Type: '{typeof(ViewComponentContainer.AmbiguousBase)}' - " +
+                "Name: 'Ambiguous.Name'" + Environment.NewLine +
+                $"Type: '{typeof(ViewComponentContainer.DerivedAmbiguous)}' - " +
```

Comment:
Might want to double check in product code that we use a reasonable sort order when building this string. Otherwise we'll start seeing flakiness on other platforms/mono.